### PR TITLE
[Dev-Bug] - Validate Deleted Files crashes when using flag -g

### PR DESCRIPTION
**Pull request by MLainer1**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/47503

## Description
Validate Deleted Files crashes when using flag -g due to git returning file_paths as PosixPath

## Screenshots
image

## Must have
- [ ] Tests
- [ ] Documentation
- [ ] [VS Code Extension](https://docs.gitlab.com/ee/ci/yaml/#tags)
  - [ ] Functionality implemented in the extension - <u>\<link-to-pr-here\></u>
  - [ ] N/A - Functionality cannot be implemented in the extension because <u>\<add-reason-here\></u>

### CODE REVIEW

File: demisto_sdk/commands/validate/tests/validators_test.py [hunk 1]
```diff
@@ -1802,6 +1802,7 @@ def test_job_unexpected_field_values_in_non_feed_job(repo, capsys,
 @pytest.mark.parametrize('file_set,expected_output,expected_result,added_files',
                          (({'mock_file_description.md'}, "[BA115]", False, set()),
                           (set(), "", True, set()),
+                          ({PosixPath('doc_files/image.png')}, "", True, {'added_file.yml'}),
```
Comment (dorschw) [inline demisto_sdk/commands/validate/tests/validators_test.py:1805]:
```suggestion
                          ({Path('doc_files/image.png')}, "", True, {'added_file.yml'}),
```

File: demisto_sdk/commands/validate/tests/validators_test.py [hunk 2]
```diff
@@ -78,7 +78,7 @@
     INCIDENT_FIELD
 from TestSuite.pack import Pack
 from TestSuite.test_tools import ChangeCWD
-
+from pathlib import PosixPath
```
Comment (dorschw) [inline demisto_sdk/commands/validate/tests/validators_test.py:81]:
```suggestion
from pathlib import Path
```

File: demisto_sdk/commands/validate/validate_manager.py [hunk 3]
```diff
@@ -1198,6 +1198,8 @@ def was_file_renamed_but_labeled_as_deleted(file_path, added_files):
 
         """
         if added_files:
+            if not isinstance(file_path, str):
+                file_path = file_path.as_posix()
```
Comment (dorschw) [inline demisto_sdk/commands/validate/validate_manager.py:1202]:
```suggestion
            file_path = str(file_path)
```

File: demisto_sdk/commands/validate/validate_manager.py [hunk 4]
```diff
@@ -1214,10 +1216,12 @@ def validate_deleted_files(self, deleted_files, added_files) -> bool:
 
         is_valid = True
         for file_path in deleted_files:
+            if not isinstance(file_path, str):
+                file_path = file_path.as_posix()
```
Comment (dorschw) [inline demisto_sdk/commands/validate/validate_manager.py:1220]:
```suggestion
            file_path = str(file_path)
```

**Review by dorschw — CHANGES_REQUESTED** [review]
please make sure to add a UT as well

File: demisto_sdk/commands/validate/tests/validators_test.py (hunk 2 above)
Comment (MLainer1) [reply demisto_sdk/commands/validate/tests/validators_test.py:81]: It needs to specifically be PosixPath as that is the type that is returned from Git

Comment (dorschw) [reply demisto_sdk/commands/validate/tests/validators_test.py:81]: this is the type returned from git on MacOS/UNIX machines. To ease usage on other OSes, we better use `Path`. See more [here](https://docs.python.org/3/library/pathlib.html).

**Review by dorschw — APPROVED** [review]
nice

**Review by dorschw — CHANGES_REQUESTED** [review]
Please add release notes

**Review by dorschw — APPROVED** [review]

**Pull request merged**